src/sql/platform/connection/common/connectionProfile.ts

```diff
@@ -65,10 +66,17 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
 				let capabilities = this.capabilitiesService.getCapabilities(this.providerName);
 				if (capabilities && capabilities.connection && capabilities.connection.connectionOptions) {
 					const options = capabilities.connection.connectionOptions;
+					// MSSQL Provider doesn't treat appName as special type anymore.
 					let appNameOption = options.find(option => option.specialValueType === interfaces.ConnectionOptionSpecialType.appName);
 					if (appNameOption) {
 						let appNameKey = appNameOption.name;
 						this.options[appNameKey] = Constants.applicationName;
+					} else if (this.providerName.includes(Constants.mssqlProviderName)) {
```

Comment:
I know.. not the best way to do this.. but curious to know is there a way to get this done by provider? Or should this be done for all providers instead? Not sure how many providers support appname, and if they also would receive the value from advanced options by making it non-special type; the appname option name wouldn't necessarily match so may be we could pass it back to provider to handle it..?

I didn't want to keep inconsistent connection profiles between STS and ADS Core so I went for this, but it would be good to know if there's a better way!

Reply:
I'd suggest just do inga direct comparison for the provider name being MSSQL or MSSQL-CMS, using includes like this is dangerous if another provider is added later on that has MSSQL in its name...

Took a quick look at it seems like PG and Kusto both have this special value, although I'm not 100% sure if they actually use it in any way. 

Is there a reason you don't just have app name as it is show up as an option in the advanced properties with a default of azdata, and if the user overrides it then it uses that instead? We can still add the suffix as needed. 

e.g. by default we'd have app names like this in the connection string

`azdata-GeneralConnection`
`azdata-ObjectExplorer`
etc.

and then if they assign it a new value we'll use that instead, e.g. for `MyCustomAppName`

`MyCustomAppName-GeneralConnection`
`MyCustomAppName-ObjectExplorer`

I'm just not sure why we need to keep azdata in it, even if they specify their own value.

Reply:
`azdata` identifier in appname is retained for our server-side telemetry purposes - as otherwise we wouldn't be able to track connections from ADS. Feature identifiers will eventually go away when pooling is enabled, but I need to do testing/validations before I remove those after enabling pooling.

Reply:
Is there a reason you need to do that here instead of in STS? That seems like a better way to guarantee that we always have it in the connection and would prevent the possibility of a bug on the ADS side causing the URIs to be different.

Reply:
Because we interchange URIs from ADS to STS - it's needed in both ADS and STS I think.